main/lsp/requests/hover.cc

```diff
@@ -48,17 +63,36 @@ LSPResult LSPLoop::handleTextDocumentHover(unique_ptr<core::GlobalState> gs, con
         }
 
         auto resp = move(queryResponses[0]);
+
+        optional<string> documentation = nullopt;
+        if (
+                resp->isIdent() || 
+                resp->isLiteral() || 
+                resp->isConstant() || 
+                resp->isDefinition()) {
+            auto origins = resp->getTypeAndOrigins().origins;
+            if (!origins.empty()) {
+                auto loc = origins[0];
+                documentation = findDocumentation(loc.file().data(*gs).source(), loc.beginPos());
```

Comment:
`loc` might not exist, so you should check `if loc.exists()` before calling `loc.file().data(*gs)`.

Sorbet uses locs with file ID 0 to indicate invalid locations internally, and calling `data()` on them will cause an exception in debug binaries and a memory error in production binaries.

Reply:
Ah, didn't realise that. Thanks! Guarded against this and the sendResp receiver in bcb83f653